hyperspy/_signals/eds.py

```diff
@@ -996,6 +1006,8 @@ def _add_xray_lines_markers(self, xray_lines):
                 x=line_energy[i], y=intensity[i] * 1.1, text=xray_lines[i],
                 rotation=90)
             self.add_marker(text)
+            line.events.closed.connect(partial(self._remove_xray_lines_markers,
+                                               xray_lines[i]), [])
```

Comment:
Actually, looking over the diff, I think this code is incorrect. `_remove_xray_lines_markers()` expects a collection on markers. Maybe the easiest would be to have:
```python
line.events.closed.connect(lambda marker: self._remove_xray_lines_markers([marker]))
```
Note, this is simply by the diff, I haven't actually tested any of the above.